Add ubuntu22.04 build/test docker containers

leborchuk opened the pull request:
This is the copy of https://github.com/apache/cloudberry-devops-release/pull/13 

We decided to move cloudberry-devops-release/ to the main repo and then add support of ubuntu 22.04 here.

In these PR I add build/test docker containers that is needed for building debian binary. There is some changes compared to RPM-based builds:
1. Debian packages (and docker containers) does not need to build libxerces since 22.04 repo contains libxerces-c3.2. libxerces-c3.2 fits perfectly for cloudberry. 
2. Debian prohibits install something inside /usr/local/. So we need to build our binaries somewhere else, then list relative local build paths in control file and finally write install script to copy it in /usr/local/cloudberry-db. In reality we just create symbolic link /usr/cloudberry-db -> /usr/local/cloudberry-db

After successful merging, I will add GitHub Actions scripts for creating Debian packages. See the example here https://github.com/leborchuk/cloudberrydb/actions/runs/15662457461/job/44122150152

tuhaihe reviewed (approved):
+1. Hope we can merge it after #1354.

Also:
- can revise the commit message following the [.gitmessage template](https://github.com/apache/cloudberry/blob/main/.gitmessage).
- can also update the commit author for your contribution track, not big deal, FYI :)

tuhaihe commented:
Hey @leborchuk , there are some conflicts that need to be resolved. PTAL, thanks!

leborchuk commented:
> Hey @leborchuk , there are some conflicts that need to be resolved. PTAL, thanks!

Hi @tuhaihe ! I have rebased and rewritten commit message. Could be checked again.

tuhaihe commented:
> > Hey @leborchuk , there are some conflicts that need to be resolved. PTAL, thanks!
> 
> Hi @tuhaihe ! I have rebased and rewritten commit message. Could be checked again.

LGTM now. Thanks @leborchuk !
